dd-java-agent/instrumentation/freemarker-2.3.24/src/main/java/freemarker/core/DollarVariable24DatadogAdvice.java

```diff
@@ -0,0 +1,44 @@
+package freemarker.core;
+
+import datadog.trace.api.iast.InstrumentationBridge;
+import datadog.trace.api.iast.Sink;
+import datadog.trace.api.iast.VulnerabilityTypes;
+import datadog.trace.api.iast.sink.XssModule;
+import freemarker.template.TemplateException;
+import net.bytebuddy.asm.Advice;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class DollarVariable24DatadogAdvice {
+
+  protected static final Logger log = LoggerFactory.getLogger(DollarVariable24DatadogAdvice.class);
+
+  public static class DollarVariableAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    @Sink(VulnerabilityTypes.XSS)
+    public static void onEnter(
+        @Advice.Argument(0) final Environment environment, @Advice.This final DollarVariable self) {
+      if (environment == null || self == null) {
+        return;
+      }
+      final XssModule xssModule = InstrumentationBridge.XSS;
+      if (xssModule == null) {
+        return;
+      }
+      if (DollarVariable24Helper.fetchAutoEscape(self)) {
+        return;
+      }
+      String charSec = null;
+      try {
+        charSec = (String) self.calculateInterpolatedStringOrMarkup(environment);
+      } catch (TemplateException e) {
+        log.debug("Failed to get DollarVariable templateModel", e);
```

Comment:
```suggestion
        xssModule.onUnexpectedException("Failed to get DollarVariable templateModel", e);
```

And you can remove the logger declaration and also the advice as a helper.

Reply:
The code was moved to the helper in order to avoid to use the logger in the advice